### PR TITLE
RFC8183 publisher request repository response

### PR DESCRIPTION
**Pull request by erikrozendaal**
Implementation of RFC8183 `publisher_request` and `repository_response` classes and XML mapping.

 * [ ] Unit tests
 * [ ] I have updated the changelog in README.md

### CODE REVIEW

**Review by wibisono — APPROVED** [review]

File: src/main/java/net/ripe/rpki/commons/provisioning/identity/RepositoryResponse.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    Optional<URI> rrdpNotificationUri;
+    ProvisioningIdentityCertificate repositoryBpkiTa;
+
+    /**
+     * Create a child identity to offer to your parent, including a suggested handle. Note that
+     * your parent may ignore this handle!
+     */
+    public RepositoryResponse(Optional<String> tag, URI serviceUri, String publisherHandle, URI siaBase,
+                              Optional<URI> rrdpNotificationUri, ProvisioningIdentityCertificate repositoryBpkiTa) {
+        this.tag = Objects.requireNonNull(tag, "tag must not be null");
```
Comment (ties) [inline src/main/java/net/ripe/rpki/commons/provisioning/identity/RepositoryResponse.java:34]: For a few of those: `Strings.isNullOrEmpty` may be better.

File: src/main/java/net/ripe/rpki/commons/provisioning/identity/PublisherRequestSerializer.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            final Element root = getElement(doc, "publisher_request")
+                    .orElseThrow(() -> new IdentitySerializerException("publisher_request element not found"));
+
+            final Optional<String> tag = getAttributeValue(root, "tag");
+
+            final String publisherHandle = getRequiredAttributeValue(root, "publisher_handle");
+
+            final ProvisioningIdentityCertificate publisherBpkiTa =
+                getBpkiElementContent(doc, "publisher_bpki_ta")
+                    .map(bpkiTa -> getProvisioningIdentityCertificate(bpkiTa))
```
Comment (ties) [inline src/main/java/net/ripe/rpki/commons/provisioning/identity/PublisherRequestSerializer.java:46]: Implicit: base64 decode happens there.

File: src/main/java/net/ripe/rpki/commons/provisioning/identity/RepositoryResponseSerializer.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            final String publisherHandle = getRequiredAttributeValue(root, "publisher_handle");
+
+            final URI siaBase = URI.create(getRequiredAttributeValue(root, "sia_base"));
+
+            final Optional<URI> rrdpNotificationUri = getAttributeValue(root, "rrdp_notification_uri")
+                .map(URI::create);
+
+            final ProvisioningIdentityCertificate repositoryBpkiTa =
+                getBpkiElementContent(doc, "repository_bpki_ta")
+                    .map(bpkiTa -> getProvisioningIdentityCertificate(bpkiTa))
```
Comment (ties) [inline src/main/java/net/ripe/rpki/commons/provisioning/identity/RepositoryResponseSerializer.java:48]: Same

**Review by ties — CHANGES_REQUESTED** [review]
Probably better to also check for empty strings for the values that are actually really require. Will go over those now.

File: src/main/java/net/ripe/rpki/commons/provisioning/identity/PublisherRequestSerializer.java (hunk 2 above)
Comment (lolepezy) [reply src/main/java/net/ripe/rpki/commons/provisioning/identity/PublisherRequestSerializer.java:46]: It throws `IllegalArgumentException ` from decode which is caught here, so I guess, it's ok.

**Review by lolepezy — APPROVED** [review]

File: src/main/java/net/ripe/rpki/commons/provisioning/identity/PublisherRequestSerializer.java (hunk 2 above)
Comment (ties) [reply src/main/java/net/ripe/rpki/commons/provisioning/identity/PublisherRequestSerializer.java:50]: Yeap. I was not sure it checked that it was valid base64 but it does so implicitly

File: src/main/java/net/ripe/rpki/commons/provisioning/identity/RepositoryResponse.java (hunk 1 above)
Comment (ties) [reply src/main/java/net/ripe/rpki/commons/provisioning/identity/RepositoryResponse.java:34]: Scratch that.

**Review by ties — APPROVED** [review]

**Review by lolepezy — APPROVED** [review]

**Pull request merged**